dotcom-rendering/cypress/e2e/parallel-1/sign-in-gate.cy.js

```diff
@@ -188,5 +223,293 @@ describe('Sign In Gate Tests', function () {
+					'Register for free',
+				);
+			});
+
+			it('should show the main sign in gate if GU_CO_COMPLETE is present but flag is false', function () {
+				setGuCOCompleteCookie('new', 'DigitalPack');
+
+				postArticleAndScrollToGateForLazyLoad({
+					switchOverride: {
+						personaliseSignInAfterCheckoutGate: false,
+					},
+				});
+
+				cy.get('[data-cy=sign-in-gate-main]').should('be.visible');
+				cy.get('[data-cy=sign-in-gate-main]').contains(
+					'You need to register to keep reading',
+				);
+			});
+
+			it('should show the main sign in gate if GU_CO_COMPLETE is present but flag is not set', function () {
+				setGuCOCompleteCookie('new', 'DigitalPack');
+
+				postArticleAndScrollToGateForLazyLoad();
+
+				cy.get('[data-cy=sign-in-gate-main]').should('be.visible');
+				cy.get('[data-cy=sign-in-gate-main]').contains(
+					'You need to register to keep reading',
+				);
+			});
+
+			describe('Sign in gate should show personalised copy if GU_CO_COMPLETE is present', function () {
+				// HEADER TEXT
+				const SUBSCRIPTION_HEADER = 'Thank you for subscribing';
+				const SUPPORTER_HEADER = 'Thank you for your support';
+
+				// SUBHEADER TEXT
+				const SIGN_IN_PROMPT =
+					'Remember to sign in for a better experience.';
+
+				// BODY TEXT
+				const SIGN_IN_INCENTIVES_DIGITAL = [
+					'Supporter rewards – unlock the benefits of your support',
+					'Incisive analysis and original reporting direct to your inbox, with our newsletters',
+					'Get involved in the discussion – comment on stories',
+				];
+
+				const SIGN_IN_INCENTIVES_NON_DIGITAL = [
+					'Fewer interruptions',
+					'Incisive analysis and original reporting direct to your inbox, with our newsletters',
+					'Get involved in the discussion – comment on stories',
+				];
+				// BUTTON TEXT
+				const COMPLETE_REGISTRATION_BUTTON = 'Complete registration';
+				const SIGN_IN_BUTTON = 'Sign in';
+
+				it('user is new and has a digital subscription', function () {
+					setGuCOCompleteCookie('new', 'DigitalPack');
+
+					postArticleAndScrollToGateForLazyLoad({
+						switchOverride: {
+							personaliseSignInAfterCheckoutGate: true,
+						},
+					});
+
+					cy.get('[data-cy=sign-in-gate-main]').should('be.visible');
+					cy.get('[data-cy=sign-in-gate-main]').contains(
+						SUBSCRIPTION_HEADER,
+					);
+					cy.get('[data-cy=sign-in-gate-main]').contains(
+						SIGN_IN_PROMPT,
+					);
+					SIGN_IN_INCENTIVES_DIGITAL.forEach((item) => {
+						cy.get('[data-cy=sign-in-gate-main]').contains(item);
+					});
+					cy.get('[data-cy=sign-in-gate-main_register]').contains(
+						COMPLETE_REGISTRATION_BUTTON,
+					);
+					cy.get('[data-cy=sign-in-gate-main_register]')
+						.should('have.attr', 'href')
+						.and('contains', '/register?returnUrl=')
+						.and(
+							'match',
+							/componentId%3Dmain_variant_\d_personalised_new_DigitalPack/,
+						);
+				});
+
+				it('user is new and has a paper subscription', function () {
```

Comment:
This is a non-blocking suggestion but it seems like there are a few tests here that don't necessarily have to be e2e Cypress tests driving a real browser.

Could they be lighter component tests that assert on the rendered component output using react-testing-library? It might also mean you wouldn't need to use fixtures in Cypress to override switches.

Examples here:
https://github.com/guardian/dotcom-rendering/blob/9904276f6ccd7f7706dc0592ba32dad2206a799a/dotcom-rendering/src/web/components/ReaderRevenueLinks.test.tsx

https://github.com/guardian/dotcom-rendering/blob/9904276f6ccd7f7706dc0592ba32dad2206a799a/dotcom-rendering/src/web/components/MostViewedRight.test.tsx

Probably worth checking with the team or @OllysCoding if unit tests with react-testing-library is still an approved approach.

Reply:
Thanks, yeah I agree, the level of testing is probably a bit heavy for full e2e tests here. Started it this way coming in new to the code base and I feel that now the feature has been developed in this way I'm not sure it would be worth the effort to retro fit a different testing strategy. But will definitely look at testing at the component level for these kind of changes in future

Reply:
I wonder if given the way this component works and the fact it relies on both switches & cookies, at least some e2e testing is probably necessary. 

That said given how (nicely) isolated the SignInGateMainCheckout component is, it's definitely a candidate for something like react-testing library for this kind of variation testing!